src/js/domain/vendingMachine.ts

```diff
@@ -0,0 +1,146 @@
+import * as type from "../interface/vendingMachine.interface";
+
+import {
+  checkDuplicatedName,
+  checkNameLength,
+  checkValidPrice,
+  checkValidCount,
+  checkDividedByMinimumCoin,
+  checkMoneyOverMaximum,
+  checkMoneyUnderZero,
+} from "../util/validations";
+
+import SingleProduct from "./product";
+
+export default class VendingMachine implements type.IVendingMachine {
+  private static instance: VendingMachine;
+
+  private chargedMoney = 0;
+  private products = [];
+  private changes = {
+    500: 0,
+    100: 0,
+    50: 0,
+    10: 0,
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  private constructor() {}
+
+  public static getInstance() {
+    return this.instance || (this.instance = new this());
+  }
+
+  addProduct: type.IAddProduct = ({ name, price, count }) => {
+    checkDuplicatedName(this.products, name);
+    checkNameLength(name);
+    checkValidPrice(price);
+    checkValidCount(count);
+
+    const product = new SingleProduct(name, price, count);
+    this.products.push(product);
+    return product;
+  };
+
+  getProducts: type.IGetProducts = () => {
+    return this.products;
+  };
+
+  updateProduct: type.IUpdateProduct = (id, name, price, count) => {
+    checkDuplicatedName(this.products, name, id);
+    checkNameLength(name);
+    checkValidPrice(price);
+    checkValidCount(count);
+
+    const product = this.products.find((product) => {
+      return id === product.getId();
+    });
```

Comment:
```suggestion
    const product = this.products.find((product) => id === product.getId());
```
이렇게 한 줄로 표현 가능할것 같아요~!

Reply:
감사합니다 : D 반영했어요~!
[da318cd](https://github.com/woowacourse/javascript-vendingmachine/pull/61/commits/da318cd9390ecf41ef4bac14fb60001fba95d5b3)